kernel/Kconfig.init

```
@@ -6,6 +6,30 @@
 
 menu "SoC and Board Hooks"
 
+config SOC_EARLY_RESET_HOOK
+	bool "Run SoC-specific early reset hook"
+	help
+	  Run a SoC-specific hook early in the reset/startup code (__start).
+	  A custom hook soc_early_reset_hook() will be executed at the beginning
+	  of the architecture-specific startup code, after hardware has been
+	  configured (as required by CONFIG_INIT_ARCH_HW_AT_BOOT) but before
+	  architecture's own resume logic.
+
+	  Returning from this hook is not mandatory: it can be used to implement
+	  special logic to resume execution in some scenarios (for example, when
+	  a bootloader is used).
+
+	  The hook is called immediately after reset so use stack wisely if
+	  the hook intends to return to continue with arch's resume/reset activity.
+	  The stack pointer register should be considered as not initialized by the caller of
+	  this hook. In particular, this means that the hook is NOT allowed to
+	  "push" any data using this stack pointer value. However, the hook may use
+	  an implementation-specific area as stack if desired; in such case,
+	  the original value of the stack pointer needs not to be "restored"
+	  before returning control to the hook's caller.
```

Nit: align lines to similar length (+ small reword):

```suggestion
	  The stack pointer register should be considered as not initialized upon
	  call to this hook. In particular, this means that the hook is NOT allowed
	  to "push" any data using this stack pointer value. However, the hook may
	  use an implementation-specific area as stack if desired; in such case,
	  the original value of the stack pointer needs not to be "restored" before
	  returning control to the hook's caller.
```